### DISCUSSION TITLE
System Access token instead of PAT

### DISCUSSION THREAD

**Discussion by AshishNigamNoida** [rfennell/AzurePipelines#1080] [Q&A]
Hi,

Is there any way to use $env:SYSTEM_ACCESSTOKEN instead of PAT  for the richardfennellBM.BM-VSTS-WIKIUpdater-Tasks.WikiUpdaterTask.WikiUpdaterTask@1

Please help me if there is any option.

Regards,
Ashish

**Comment by rfennell** [comment]
Yes, you just need to set the `useAgentToken=true` parameter as detailed in the [wiki](https://github.com/rfennell/AzurePipelines/wiki/WIKI-PdfExport-Task) e.g:

```
 - task: richardfennellBM.BM-VSTS-WIKIUpdater-Tasks-DEV.WikiUpdaterTask.WikiUpdaterTask@1
   displayName: 'Git based WIKI Updater - Azure DevOps as Agent'
   inputs:
      repo: 'dev.azure.com/richardfennell/GitHub/_git/GitHub.wiki'
      filename: TestPageForWikiUpdater.md
      contents: |
          Header
          ---
          $(OutputText)
           ---
          Footer
       message: 'Update from Release - $(Release.ReleaseName) as Agent'
       gitname: builduser
       gitemail: 'build@demo'
       useAgentToken: true
       localpath: '$(System.DefaultWorkingDirectory)\azurerepoasagent'
```

**Reply by AshishNigamNoida** [reply]
- task: richardfennellBM.BM-VSTS-WIKIUpdater-Tasks.WikiUpdaterTask.WikiUpdaterTask@1
              displayName: 'Git based WIKI File Updater'
              inputs:
                  repo: dev.azure.com/ashish12Dec/MyProject/_git/MyProject.wiki
                  filename: wikireleasenotes.md
                  replaceFile: false
                  appendToFile: false
                  dataIsFile: true
                  sourceFile: '$(Build.ArtifactStagingDirectory)/wikireleasenotes.md'
                  message: MyNewReleaseNote
                  gitname: MyProject
                  gitemail: 'ashish@'
                  useAgentToken: true


I am using above task but getting below access issue=======

##[error]Error: Cloning into 'D:\a\1\s\repo'...
remote: TF401019: The Git repository with name or identifier MyProject.wiki does not exist or you do not have permissions for the operation you are attempting.
fatal: repository 'https://dev.azure.com/ashish12Dec/MyProject/_git/MyProject.wiki/' not found

##[error]Error: Cloning into 'D:\a\1\s\repo'...
remote: TF401019: The Git repository with name or identifier MyProject.wiki does not exist or you do not have permissions for the operation you are attempting.
fatal: repository 'https://dev.azure.com/ashish12Dec/MyProject/_git/MyProject.wiki/' not found

**Reply by rfennell** [reply]
Check the details in the wiki, at the foot of the page on urls (Azure DevOps wiki urls are strange) and permissions

I suspect the issue

- url is wrong
- the build agent is missing permissiond
